drivers/soundwire/intel_init.c

```diff
@@ -269,9 +269,11 @@ static struct sdw_intel_ctx
 
 err:
 	sdw_intel_cleanup(ctx);
-link_err:
 	driver_unregister(&intel_sdw_driver.driver);
 register_err:
+	if (ctx->links)
+		kfree(ctx->links);
```

Comment:
no check for NULL before kfree() please

Reply:
> no check for NULL before kfree() please

rtx->links could be freed and set to null in sdw_intel_cleanup(). It will get oops if we don't check for NULL before kfree()

Reply:
no, it's fine to do kfree on a NULL pointer, but it's not elegant at all. Double-frees are a sign of a flow that can be improved.